woocommerce/woocommerce-ios
# Google Ads: Add entry point for campaigns from Hub Menu

Closes: #13180 


## Description

This PR adds the entry point to Google Ads campaigns from the Hub Menu. Due to the messy logic in the hub menu, this PR also includes refactoring to avoid duplicated requests.

Changes:
- `BlazeEligibilityChecker`: Injected `site` to the checker for more explicit logic. The previous implementation using `stores` singleton makes it hard to track issues. This leads to the updates at relevant call sites.
- `HubMenuViewModel`: 
  - Check eligibility for Blaze, Google Ads, and Inbox when site info is available.
  - Updated `setupGeneralElements` to combine all the eligibility to determine the appropriate menu items.
 
## Testing steps

- Pre-requisite: Install, activate, and set up the Google Ads plugin to a test site following pe5sF9-2Qo-p2. Ensure that you're using the plugin version 2.7.5 or above.
- Log in to your test site, and confirm that there's an item in the Hub Menu for Google Ads. The wording follows Android's [PR](https://github.com/woocommerce/woocommerce-android/pull/11830).
- Switch to a site without Google Ads plugin. Confirm that the Google Ads item is not available on the Hub Menu.

Note: Tapping on the item will crash the app because we haven't handled the action yet.

## Screenshots

<img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/cc788f2b-7a45-43ab-b5e4-cc33a72e49ae" width=320 />


---
